### DISCUSSION TITLE
How to use pyaedt to iterate through all the objects and select the objects by name?

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#4359] [Q&A] [answered]
How to use pyaedt to iterate through all the objects in IcepakDesign and put the objects that start with a specific string into a new list?

**Comment by YLSHA** [comment]
`ipk.modeler.get_object_from_name('obeject_name')` with a loop
Will this be helpful

**Reply by aknsj** [reply]
Can you be more specific?How to loop?

**Reply by YLSHA** [reply]
What I mean is to create an empty list, add name to the list after creating an object. Next, use the API mentioned above to access various objects and perform operations with this list.

**Reply by aknsj** [reply]
For example, if there are a number of objects starting with "block" in the project, and I don't know the number and full name, how can I iterate through all the objects in the project? As in iron python, I can write it like this   object_name_list = [self.oEditor.GetObjectName(i) for i in range(self.oEditor.GetNumObjects())]
            objects_list = [obj for obj in object_name_list if obj.lower().startswith("block")]

**Comment by fwlfHS** [comment, marked as answer]
Does that work?:
`all_objs = ipk.modeler.object_list`
`objects_list = [obj for obj in all_objs if obj.name.lower().startswith("block")]`
`name_list = [obj.name for obj in all_objs if obj.name.lower().startswith("block")]`
